src/thunderbolt/tbtctl.c

```diff
@@ -0,0 +1,615 @@
+                log_error_errno(r, "Could not remove authorization: %m");
+                return EXIT_FAILURE;
+        }
+
+        r = tb_store_remove_device(store, uuid);
+        if (r < 0) {
+                log_error_errno(r, "Could not remove device: %m");
+                return EXIT_FAILURE;
+        }
+
+        return EXIT_SUCCESS;
+}
+
+static const struct CtlCmd cmd_forget = {
+        .name = "forget",
+        .func = forget_device,
+        .desc = "Remove a device from the database",
+        .root = true,
+};
+
+
+static const struct CtlCmd *ctrl_cmds[] = {
+        &cmd_list,
+        &cmd_authorize,
+        &cmd_forget,
+
+        &cmd_udev
+};
+
+static void help(void) {
+        unsigned int i;
+
+        printf("%s [--version] [--debug] COMMAND [OPTIONS]\n\n"
+               "Manager thunderbolt devices\n\n"
+               "  -h --help             Show this help and exit\n"
+               "  --version             Print version string and exit\n"
+               "\n"
+               "Commands:\n"
+               , program_invocation_short_name);
+
+        for (i = 0; i < ELEMENTSOF(ctrl_cmds); i++) {
+                const struct CtlCmd *cmd = ctrl_cmds[i];
+                if (!cmd->desc)
+                        continue;
+
+                printf("  %-20s  %s\n", cmd->name, cmd->desc);
+        }
+}
+
+
+#define ARG_VERSION  0x100
+#define ARG_NOROOT   0x101
+
+int main(int argc, char *argv[]) {
+        _cleanup_udev_unref_ struct udev *udev = NULL;
+        const char *cmdname;
+        const struct CtlCmd *cmd;
+        bool root_check = true;
+        static const struct option options[] = {
+                { "debug",   no_argument, NULL, 'd' },
+                { "help",    no_argument, NULL, 'h' },
+                { "version", no_argument, NULL, ARG_VERSION },
+                { "noroot",  no_argument, NULL, ARG_NOROOT  },
+                {}
+        };
+        unsigned int i;
+        int c, r;
+
+        setlocale(LC_ALL, "");
+        log_parse_environment();
+        log_open();
+
+        while ((c = getopt_long(argc, argv, "+dhV", options, NULL)) >= 0)
+                switch (c) {
+
+                case 'd':
+                        log_set_max_level(LOG_DEBUG);
+                        break;
+
+                case 'h':
+                        help();
+                        return EXIT_SUCCESS;
+
+                case ARG_VERSION:
+                        version();
+                        return EXIT_SUCCESS;
+
+                case ARG_NOROOT:
+                        root_check = false;
+                        break;
+
+                default:
+                        assert_not_reached("Unhandled option");
+                }
+
+
+        cmdname = argv[optind];
+
+        if (!cmdname) {
+                fprintf(stderr, "%s: need to specify command\n", program_invocation_short_name);
+                fprintf(stderr, "  use --help for available commands\n");
```

Comment:
please always log with log_error() and friends, never use fprintf(stderr) directly. Also, please log two lines that belong together (like these two) as one, so that only one log entry is generated should people run this program with $SYSTEMD_LOG_TARGET=journal (yeah, it's unlikely they will, but we generally follow this scheme everywhere)